storage/elasticsearch/elasticsearch.go

```diff
@@ -141,10 +144,17 @@ func (c *ElasticSearchStorage) initialize() error {
 	return nil
 }
 
-func New(addr string, port int) (*ElasticSearchStorage, error) {
+func New() (*ElasticSearchStorage, error) {
 	c := elastigo.NewConn()
-	c.Domain = addr
-	c.Port = strconv.FormatInt(int64(port), 10)
+
+	elasticonfig := config.GetConfig().Section("storage").Key("elasticsearch").Strings(":")
+	if len(elasticonfig) != 2 {
+		fmt.Fprintf(os.Stderr, "Config file is misconfigured : check elasticsearch key format\n")
```

Comment:
Since this is a library and not a binary I think you should return an error instead of exit, IHMO